pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -90,25 +79,25 @@ public FunctionMetaDataManager(WorkerConfig workerConfig,
      */
     public void initialize() {
         try {
+            initializeTailer();
             this.functionMetaDataTopicTailer = new FunctionMetaDataTopicTailer(this,
                     pulsarClient.newReader(), this.workerConfig, this.errorNotifier);
             // read all existing messages
-            this.setInitializePhase(true);
             while (this.functionMetaDataTopicTailer.getReader().hasMessageAvailable()) {
                 this.functionMetaDataTopicTailer.processRequest(this.functionMetaDataTopicTailer.getReader().readNext());
             }
-            this.setInitializePhase(false);
-            
-
         } catch (Exception e) {
             log.error("Failed to initialize meta data store", e);
-            throw new RuntimeException(e);
+            errorNotifier.triggerError(e);
         }
     }
-    
+
+    private void initializeTailer() throws PulsarClientException {
```

Comment:
Should we separate out the metadata tailer from the meta manager like what we did for the assignment tailer and runtime manager?

Reply:
As it stands now, currently PulsarMetaDataManager encapsulates all of this logic within itself. I can split this up in a following changes